### PR TITLE
Update export tool to version `1.0.0-beta.2`.

### PR DESCRIPTION
**Pull request by aaronweeden**
This updates the `xdmod-ondemand-export` tool to version `1.0.0-beta.2` with the following changes:

- The README now instructs that a new `xdmod-ondemand-export` user should be created (with no login or shell access) to run the tool.
- The tool no longer uses a Bash script to load the API token into the environment. Instead, the token is read from a read-only file.
- The tool now uses a `last-run.json` file to keep track of previously-POSTed files, rather than keeping track of this in `conf.ini`. This makes it so the `conf.ini` file is now read-only to the tool, it simplifies the log parsing code, and it fixes a false assumption in the previous version of the tool that log files are ordered by request time, which could have led to missing data in some situations.
- The `compression` configuration parameter has been removed, and the tool instead detects gzip files by their `.gz` extension, since there may be a mix of compressed and uncompressed logs, e.g., if logrotate compresses files once they reach a certain size.
- The README now instructs the user to install from PyPI rather than GitHub.
- The README now emphasizes that the configuration file should be edited before running the tool.
- The instructions are updated to encourage file permissions of:
    - `400` for `conf.ini` and `.token`.
    - `600` for `last-run.json`.
- The Python script is now packaged as a console script to simplify the command line for running it and reduce the number of files in the package.
- The package's long description is now set to link to the README on GitHub so that the README can be updated without uploading a new version to PyPI.
- Code coverage is now tested by CircleCI.
- The README instructions are now tested by CircleCI.
- A `development.md` documentation file has been added. It currently documents how to make a new version and distribute it to PyPI.
- Minimum dependency version numbers have now been specified in the `setup.cfg` file.

### CODE REVIEW

File: tools/xdmod-ondemand-export/xdmod_ondemand_export/xdmod_ondemand_export.py [hunk 1]
```diff
@@ -24,30 +24,29 @@ def __init__(self):
         # Override how exceptions are handled.
         sys.excepthook = self.__excepthook
         self.__logger.debug('Using arguments: ' + str(self.__args))
-        self.__validate_file_permissions()
+        self.__validate_file_permissions_600(
+            os.path.expanduser('~/xdmod-ondemand-export/.token')
```
Comment (jpwhite4) [inline tools/xdmod-ondemand-export/xdmod_ondemand_export/xdmod_ondemand_export.py:28]: Not sure its a good idead to hard code the `~/xdmod-ondemand-export` path. The README doesn't make it clear that you _have_ to use ~/xdmod-ondemand-export as the package dir and it won't work if you choose something different.

You can use python to tell you where the package resources have been installed - thus not hardcoding it:

See example code at:

https://github.com/ubccr/supremm/blob/master/src/supremm/config.py#L62

Comment (jpwhite4) [reply tools/xdmod-ondemand-export/xdmod_ondemand_export/xdmod_ondemand_export.py:28]: Another alternative is to have a 'default' value in the software (like you have) but also add a command line argument to override the default.

Comment (jpwhite4) [reply tools/xdmod-ondemand-export/xdmod_ondemand_export/xdmod_ondemand_export.py:28]: Also I don't think we want to try to enforce 600 permissions for the token file. It only needs read-only for the user account that runs the process.

Comment (aaronweeden) [reply tools/xdmod-ondemand-export/xdmod_ondemand_export/xdmod_ondemand_export.py]: I have added a command line argument to override the token path.

The script prints warnings if the file permissions are not 600 on the token and/or JSON file, but it goes ahead and runs anyway. Should we remove those warnings?

**Review by jpwhite4 — APPROVED** [review]

**Pull request merged**